doc/mgmt_netconf/hld_NETCONF-NTT.md

```diff
@@ -0,0 +1,790 @@
+  - Delete config via NETCONF
+  - Confirm `CONFIG_DB` entries are removed
+- get-config 
+  - Send NETCONF request to get configuration (filter: subtree interfaces)
+  - Confirm interface config is retured as expected
+
+#### Config Items
+
+Config items to test using interface address config.
+
+Expected CONFIG_DB entry
+
+Confirm config was applied.
+
+```
+admin@sonic:~$ sonic-db-cli CONFIG_DB keys INTERFACE*
+INTERFACE|Ethernet0
+INTERFACE|Ethernet0|10.1.1.1/24
+
+admin@sonic:~$ show ip int
+Interface    Master    IPv4 address/mask    Admin/Oper    BGP Neighbor    Neighbor IP
+-----------  --------  -------------------  ------------  --------------  -------------
+Ethernet0              10.1.1.1/24          up/up         N/A             N/A
+docker0                240.127.1.1/24       up/down       N/A             N/A
+eth0                   192.168.122.169/24   up/up         N/A             N/A
+lo                     127.0.0.1/16         up/up         N/A             N/A
+```
+
+Test script (in Python)
+
+```
+from ncclient import manager
+
+m = manager.connect(
+    host="192.168.122.169",
+    port="830",
+    timeout=30,
+    username="admin",
+    password="YourPaSsWoRd",
+    hostkey_verify=False,
+)
+try:
+    m.edit_config(config=cfg, format="xml", default_operation="merge", target="running")
+finally:
+    m.close_session()
+```
+
+NETCONF edit-config (merge) payload
+
+```
+<config xmlns="urn:ietf:params:xml:ns:netconf:base:1.0">
+  <interfaces xmlns="http://openconfig.net/yang/interfaces">
+    <interface>
+      <name>Ethernet0</name>
+      <config>
+        <name>Ethernet0</name>
+      </config>
+      <subinterfaces>
+        <subinterface>
+          <index>0</index>
+          <config>
+            <index>0</index>
+          </config>
+          <ipv4 xmlns="http://openconfig.net/yang/interfaces/ip">
+            <addresses>
+              <address>
+                <ip>10.1.1.1</ip>
+                <config>
+                  <ip>10.1.1.1</ip>
+                  <prefix-length>24</prefix-length>
+                </config>
+              </address>
+            </addresses>
+          </ipv4>
+        </subinterface>
+      </subinterfaces>
+    </interface>
+  </interfaces>
+</config>
+```
+
+Example of get-config
+
+```
+> print(m.get_config(source="running", filter=("subtree", interfaces)))
+
+<rpc-reply xmlns="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="urn:uuid:4e02f166-fc2d-4701-a7a1-b15145cc01d6"><data><interfaces xmlns="http://openconfig.net/yang/interfaces"><interface><name>Ethernet0</name><config><name>Ethernet0</name></config><subinterfaces><subinterface><index>0</index><config><index>0</index></config><ipv4 xmlns="http://openconfig.net/yang/interfaces/ip"><addresses><address><ip>10.1.1.1</ip><config><ip>10.1.1.1</ip><prefix-length>24</prefix-length></config></address></addresses></ipv4></subinterface></subinterfaces></interface></interfaces></data></rpc-reply>
+```
+
+## Open/Action items - if any
```

Comment:
It would be good to mention the notification support (https://datatracker.ietf.org/doc/html/rfc8639) in case for scaling up capabilities in the future.